nixos/modules/system/boot/initrd-network.nix

```diff
@@ -76,13 +75,26 @@ in
       '';
     };
 
+    boot.initrd.network.udhcpc.enable = mkOption {
+      type = types.bool;
+      default = config.networking.useDHCP || dhcpInterfaces != [];
```

Comment:
```suggestion
      default = config.networking.useDHCP || dhcpInterfaces != [];
      defaultText = literalExpression ''config.networking.useDHCP || dhcpInterfaces != []'';
```
otherwise this default will show up as a literal in the manual

Reply:
Using the expression doesn't really explain the default, though, I'd need to substitute in the definition of `dhcpInterfaces` for it to be understood, and then it would be quite unwieldy.

The current manual page has my explanation of the default in the description already, and gives the default as it would be on an entirely default NixOS configuration, which seems fine:
![Screenshot of entry in manual page](https://0x0.st/-7gg.png)

Reply:
in the context of the description we think that `dhcpInterfaces` makes sufficient sense here. and while the default *is* correct for an entirely unconfigured nixos, it does give the impression that the default is the *literal* `true`, which is somewhat misleading. it'd be good for options that default to expressions depending on the configuration to somehow communicate that they do, instead of just using their unconfigured value for the manual.

(also we're trying to make the option doc build a lot faster, and doing so requires that options with non-literal defaults have a literal defaultText.)

Reply:
> (also we're trying to make the option doc build a lot faster, and doing so requires that options with non-literal defaults have a literal defaultText.)
Ah, OK.